Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -37,7 +37,7 @@ final class CacheAdvanceTests: XCTestCase {
     // MARK: Behavior Tests
 
     func test_messages_canReadEmptyCacheThatDoesNotOverwriteOldestMessages() throws {
-        let cache = try CacheAdvance<String>(
+        let cache = try CacheAdvance<TestableMessage>(
```

Comment:
In order to get tests passing on the older platforms, I needed to encode an object other than `String`. Trying to encode the `String` would lead to the error:
> The data couldn’t be written because it isn’t in the correct format.

You can see a sample of this error (before I figured out how to fix it 😅) in [this build](https://travis-ci.com/dfed/CacheAdvance/jobs/269437929#L1531).